scene/main/viewport.cpp

```diff
@@ -3685,6 +3689,14 @@ bool SubViewport::is_using_xr() {
 
 void SubViewport::set_size(const Size2i &p_size) {
 	_set_size(p_size, _get_size_2d_override(), Rect2i(), _stretch_transform(), true);
+#ifdef TOOLS_ENABLED
+	if (get_parent()) {
+		SubViewportContainer *c = Object::cast_to<SubViewportContainer>(get_parent());
+		if (c) {
+			c->minimum_size_changed();
+		}
+	}
+#endif
```

Comment:
What is the reason for making this work in editor builds only? I already see bug reports mentioning that `SubViewportContainer` isn't resized/changed in `release` builds in the future. 😮

I think `TOOLS_ENABLED` should only be used if it's only relevant for the editor, like debug drawing.

Reply:
This only matters if you dynamically resize viewport during runtime, which sounds very rare (and we'd already have a report like that if anyone encountered this issue...). I just thought this is only relevant for the editor, although maybe there's no gain in disabling it 🤔

Reply:
> This only matters if you dynamically resize viewport during runtime

For example, game minimaps could be resized (like twice as big, or twice as small).

From what I've seen, people do manage to create non-game applications and editors, so things like that would be used relatively often there, but not necessarily for games I think. At most, think about built-in level editors.